pkg/transportrequest/cts/upload.go

```diff
@@ -189,6 +192,10 @@ func getFioriDeployStatement(
 		log.Entry().Debug("No application package found in piper config.")
 	}
 	if len(app.Name) > 0 {
+		re := regexp.MustCompile(pattern)
+		if !re.MatchString(app.Name) {
+			fmt.Errorf("application name '%s' contains spaces or special characters", app.Name)
```

Comment:
Maybe we can include the regex we check against in the error message. Then the caller knows what exactly failed. But it would be ok for me to merge that also in the current state.